### PR TITLE
SVG bug fix v1.5.x

### PR DESCRIPTION
**Pull request by dancingfrog**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

This commit fixes a urlResolve error which is produced when AngularJS is directly loaded/called by a native SVG file.

**What is the current behavior? (You can also link to an open issue here)**
 #12484
When used directly in an SVG file, AngularJS does not properly parse URL components. This behavior can be demonstrated by simply loading the current standard (bower) distribution of AngularJS in to a simple "hello angular" SVG example, like this: 

https://github.com/Uni-Sol/svg-demos/blob/23e32783493198ef39d0a9415f834c779d3b365e/data/hello-angular.svg
svg-url_resolve-bug

**What is the new behavior (if this is a feature change)?**

AngularJS properly resolves URL parts when loaded by an SVG file.

**Does this PR introduce a breaking change?**

Nope.

**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: src/ng/urlUtils.js [hunk 1]
```diff
@@ -55,31 +55,37 @@ var originUrl = urlResolve(window.location.href);
  *   | pathname      | The pathname, beginning with "/"
  *
  */
-function urlResolve(url) {
-  var href = url;
+function urlResolve(url, base) {
```
Comment (campersau) [inline src/ng/urlUtils.js]: `base` is never used.

**Comment by Narretz** [comment]
Thanks for the PR. A few things:
* I don't think we officially support running angular in SVG documents only. We do support embedded SVG in HTML documents.
* It's better to create a new branch and don't put your commits directly on the v1.5.x branch. This way you don't have to keep the branch up-to-date with origin
* You've introduced white space errors that make it harder to read the diff correctly
* Current tests are broken and there are not tests for this change

**Comment by dancingfrog** [comment]
Thanks for reviewing this, @Narretz 

There actually is already a url spec test, which this commit is passing: 
https://github.com/angular/angular.js/blob/v1.5.x/test/ng/browserSpecs.js#L272

Which functional tests are failing for you?

After I made the commit I ran `grunt test` and saw the white space "errors". I've fixed them locally, but not sure if this pull request is worth pursuing if you don't plan to support native SVG. I do have my own branch for these changes, 'real/v1.5.x-svg', into which I pulled 'origin/v1.5.x' but it includes some extras that you would not want merged in with pull request.

**Comment by Narretz** [comment]
@dancingfrog 
I didn't want to shoot down the PR from the start, just point where we are right now. So you are saying you had to make some other changes to let Angular run inside SVG documents?

The other tips are also interesting if you contribute other features / fixes.
So you have forked angular and then applied your changes to the v1.5.x branch in your fork. What I meant is that you should branch from v1.5.x and use that  new branch for your pull requests so you don't have to merge angularjs v1.5.x into your fork when you need to update

The tests that are failing can be seen in the travis results: https://travis-ci.org/angular/angular.js/builds/124075325 (We are running on different browsers)

**Comment by dancingfrog** [comment]
I'm trying to get through the full suite of tests but it looks like some of the the grunt tasks need to be updated. For instance, "protractor:normal" is relying on remote resources that no longer exists:
```
Running "protractor:normal" (protractor) task
Starting selenium standalone server...
[launcher] Running 1 instances of WebDriver
Selenium standalone server started at http://192.168.44.9:50852/wd/hub
...................................................................................................................................................................................................................................................................................................
browser console errors: [ { level: { name: 'SEVERE', value: 1000 },
    message: 'http://www.google-analytics.com/ga.js 0:0 Failed to load resource: net::ERR_NAME_NOT_RESOLVED',
    timestamp: 1461116036547,
    type: '',
    closure_uid_122338954: undefined } ]
F
browser console errors: [ { level: { name: 'SEVERE', value: 1000 },
    message: 'http://www.google-analytics.com/ga.js 0:0 Failed to load resource: net::ERR_NAME_NOT_RESOLVED',
    timestamp: 1461116041660,
    type: '',
    closure_uid_122338954: undefined } ]
F
browser console errors: [ { level: { name: 'SEVERE', value: 1000 },
    message: 'http://www.google-analytics.com/ga.js 0:0 Failed to load resource: net::ERR_NAME_NOT_RESOLVED',
    timestamp: 1461116044570,
    type: '',
    closure_uid_122338954: undefined } ]
F
browser console errors: [ { level: { name: 'SEVERE', value: 1000 },
    message: 'http://www.google-analytics.com/ga.js 0:0 Failed to load resource: net::ERR_NAME_NOT_RESOLVED',
    timestamp: 1461116047744,
    type: '',
    closure_uid_122338954: undefined } ]
F
browser console errors: [ { level: { name: 'SEVERE', value: 1000 },
    message: 'http://www.google-analytics.com/ga.js 0:0 Failed to load resource: net::ERR_NAME_NOT_RESOLVED',
    timestamp: 1461116050529,
    type: '',
    closure_uid_122338954: undefined } ]
F
browser console errors: [ { level: { name: 'SEVERE', value: 1000 },
    message: 'http://www.google-analytics.com/ga.js 0:0 Failed to load resource: net::ERR_NAME_NOT_RESOLVED',
    timestamp: 1461116053149,
    type: '',
    closure_uid_122338954: undefined } ]
F
```

... so I get a bunch of 404s that are followed by test failures. I don't know if the subsequent tests are failing because the previous task bailed out or because of a genuine flaw in the tiny bit of code that I updated. Please have a look at the diff and let me know if it has anything to do with these test failure messages:
```
Failures:

  1) docs.angularjs.org App should change the page content when clicking a link to a service
  at 651.558s [Wed, 20 Apr 2016 01:34:01 GMT]
   Message:
     Expected 1 to equal 0.
   Stacktrace:
     Error: Expected 1 to equal 0.
    at new jasmine.ExpectationResult (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:137:32)
    at [object Object].<anonymous> (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:1349:29)
    at [object Object].toEqual (R:\Projects\Currents\angular\node_modules\protractor\node_modules\jasminewd\index.js:248:11)
    at R:\Projects\Currents\angular\docs\app\e2e\app.scenario.js:19:34
    at Array.forEach (native)
    at goog.async.run.processWorkQueue (R:\Projects\Currents\angular\node_modules\protractor\node_modules\selenium-webdriver\lib\goog\async\run.js:130:15)

  2) docs.angularjs.org App should be resilient to trailing slashes
  at 654.64s [Wed, 20 Apr 2016 01:34:04 GMT]
   Message:
     Expected 1 to equal 0.
   Stacktrace:
     Error: Expected 1 to equal 0.
    at new jasmine.ExpectationResult (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:137:32)
    at [object Object].<anonymous> (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:1349:29)
    at [object Object].toEqual (R:\Projects\Currents\angular\node_modules\protractor\node_modules\jasminewd\index.js:248:11)
    at R:\Projects\Currents\angular\docs\app\e2e\app.scenario.js:19:34
    at Array.forEach (native)
    at goog.async.run.processWorkQueue (R:\Projects\Currents\angular\node_modules\protractor\node_modules\selenium-webdriver\lib\goog\async\run.js:130:15)

  3) docs.angularjs.org App should be resilient to trailing "index"
  at 657.502s [Wed, 20 Apr 2016 01:34:06 GMT]
   Message:
     Expected 1 to equal 0.
   Stacktrace:
     Error: Expected 1 to equal 0.
    at new jasmine.ExpectationResult (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:137:32)
    at [object Object].<anonymous> (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:1349:29)
    at [object Object].toEqual (R:\Projects\Currents\angular\node_modules\protractor\node_modules\jasminewd\index.js:248:11)
    at R:\Projects\Currents\angular\docs\app\e2e\app.scenario.js:19:34
    at Array.forEach (native)
    at goog.async.run.processWorkQueue (R:\Projects\Currents\angular\node_modules\protractor\node_modules\selenium-webdriver\lib\goog\async\run.js:130:15)

  4) docs.angularjs.org App should be resilient to trailing "index/"
  at 660.488s [Wed, 20 Apr 2016 01:34:09 GMT]
   Message:
     Expected 1 to equal 0.
   Stacktrace:
     Error: Expected 1 to equal 0.
    at new jasmine.ExpectationResult (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:137:32)
    at [object Object].<anonymous> (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:1349:29)
    at [object Object].toEqual (R:\Projects\Currents\angular\node_modules\protractor\node_modules\jasminewd\index.js:248:11)
    at R:\Projects\Currents\angular\docs\app\e2e\app.scenario.js:19:34
    at Array.forEach (native)
    at goog.async.run.processWorkQueue (R:\Projects\Currents\angular\node_modules\protractor\node_modules\selenium-webdriver\lib\goog\async\run.js:130:15)

  5) docs.angularjs.org App should display formatted error messages on error doc pages
  at 663.074s [Wed, 20 Apr 2016 01:34:12 GMT]
   Message:
     Expected 1 to equal 0.
   Stacktrace:
     Error: Expected 1 to equal 0.
    at new jasmine.ExpectationResult (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:137:32)
    at [object Object].<anonymous> (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:1349:29)
    at [object Object].toEqual (R:\Projects\Currents\angular\node_modules\protractor\node_modules\jasminewd\index.js:248:11)
    at R:\Projects\Currents\angular\docs\app\e2e\app.scenario.js:19:34
    at Array.forEach (native)
    at goog.async.run.processWorkQueue (R:\Projects\Currents\angular\node_modules\protractor\node_modules\selenium-webdriver\lib\goog\async\run.js:130:15)

  6) docs.angularjs.org App should display an error if the page does not exist
  at 665.875s [Wed, 20 Apr 2016 01:34:15 GMT]
   Message:
     Expected 1 to equal 0.
   Stacktrace:
     Error: Expected 1 to equal 0.
    at new jasmine.ExpectationResult (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:137:32)
    at [object Object].<anonymous> (R:\Projects\Currents\angular\node_modules\protractor\node_modules\minijasminenode\lib\jasmine-1.3.1.js:1349:29)
    at [object Object].toEqual (R:\Projects\Currents\angular\node_modules\protractor\node_modules\jasminewd\index.js:248:11)
    at R:\Projects\Currents\angular\docs\app\e2e\app.scenario.js:19:34
    at Array.forEach (native)
    at goog.async.run.processWorkQueue (R:\Projects\Currents\angular\node_modules\protractor\node_modules\selenium-webdriver\lib\goog\async\run.js:130:15)

Finished in 665.906 seconds
297 tests, 763 assertions, 6 failures

Shutting down selenium standalone server.
[launcher] 0 instance(s) of WebDriver still running
[launcher] chrome #1 failed 6 test(s)
[launcher] overall: 6 failed spec(s)
[launcher] Process exited with error code 1
Warning: Protractor test(s) failed. Exit code: 1 Use --force to continue.

Aborted due to warnings.
```

**Comment by gkalpak** [comment]
There must be some other problem, because the tests run fine for me :confused: 

`net::ERR_NAME_NOT_RESOLVED` indicates there must be some DNS-related thing.
I would try clearing Chrome's DNS cache (from `chrome://net-internals/#dns`).

**Comment by gkalpak** [comment]
The tests are failing and there are no tests for the new behavior.
Closing, but feel free to reopen when/if you fix these issues.

**Pull request closed without merging**